pcsx2/GS/Renderers/HW/GSRendererHW.cpp

```diff
@@ -2680,6 +2680,7 @@ void GSRendererHW::Draw()
 				src->m_from_target->UnwrappedEndBlock() > m_cached_ctx.FRAME.Block()) ||
 				(m_cached_ctx.FRAME.Block() < src->m_from_target_TEX0.TBP0 && draw_end > src->m_from_target_TEX0.TBP0));
 
+			const bool sprite_or_quad = draw_sprite_tex || (m_vt.m_primclass == GS_TRIANGLE_CLASS && ((m_index.tail % 6) == 0 && TrianglesAreQuads(true)));
```

Comment:
I'm a little concerned about the performance impact of doing this on every draw. I know it's messier, but if you moved this directly into the condition on line 2694, it'd at least short circuit exit in most cases.

Reply:
Alright, I've rearranged it so it's the very last thing it attempts to check when seeing if it's a shuffle